secret-store/src/key_server_cluster/math.rs

```diff
@@ -28,6 +28,12 @@ pub struct EncryptedSecret {
 	pub encrypted_point: Public,
 }
 
+/// Calculate the inversion of a Secret key (in place) using the `libsecp256k1` crate.
```

Comment:
This is probably the most contentious change in the PR: in order to use the upstream secp lib we need a way to calculate the inversion. Pulling in a second secp256k1 crate to do that isn't very elegant.

Reply:
as long as we can make libsecp256k1 optional under `secretstore` feature it's fine